sample_project/Assets/SampleViewer/Samples/Geometry/Geometries.cs

```diff
@@ -0,0 +1,499 @@
+
+			featurePoints.Add(nextInterpolation);
+			previousInterpolation = nextInterpolation.transform.position;
+		}
+	}
+
+	// Set height for point transform and location component.
+	private void SetElevation(GameObject stop)
+	{
+		// Start the raycast in the air at an arbitrary point to ensure it is above the ground.
+		var position = stop.transform.position;
+		var raycastStart = new Vector3(position.x, position.y + RaycastHeight, position.z);
+		if (Physics.Raycast(raycastStart, Vector3.down, out RaycastHit hitInfo))
+		{
+			var elevatedPosition = hitInfo.point + new Vector3(0, MarkerHeight, 0);
+
+			stop.transform.position = elevatedPosition;
+
+			var location = stop.GetComponent<ArcGISLocationComponent>();
+			location.Position = arcGISMapComponent.EngineToGeographic(elevatedPosition);
+		}
+	}
+
+
+	private void RenderLine(ref List<GameObject> featurePoints)
+	{
+		var allPoints = new List<Vector3>();
+
+		foreach (var point in featurePoints)
+		{
+			if (point.transform.position.Equals(Vector3.zero))
+			{
+				Destroy(point);
+				continue;
+			}
+			allPoints.Add(point.transform.position);
+		}
+
+		if (isPolygonMode)
+		{
+			//add the first point to line renderer so that polygon can be closed
+			allPoints.Add(allPoints[0]);
+			foreach (var point in lastToStartInterpolationPoints)
+			{
+				if (point.transform.position.Equals(Vector3.zero))
+				{
+					Destroy(point);
+					continue;
+				}
+				allPoints.Add(point.transform.position);
+			}
+		}
+
+		lineRenderer.positionCount = allPoints.Count;
+		lineRenderer.SetPositions(allPoints.ToArray());
+	}
+
+	public void ClearLine()
+	{
+		foreach (var stop in stops)
+		{
+			Destroy(stop);
+		}
+
+		foreach (var point in featurePoints)
+		{
+			Destroy(point);
+		}
+
+		foreach (var point in lastToStartInterpolationPoints)
+		{
+			Destroy(point);
+		}
+
+		featurePoints.Clear();
+		stops.Clear();
+
+		if (lineRenderer)
+		{
+			lineRenderer.positionCount = 0;
+		}
+
+		calculation = 0;
+
+		UpdateDisplay();
+	}
+
+	public void ResetMode()
+	{
+		ClearLine();
+
+		isPolygonMode = false;
+		isEnvelopeMode = false;
+		isPolylineMode = false;
+
+		ModeButtons[0].interactable = true;
+		ModeButtons[1].interactable = true;
+		ModeButtons[2].interactable = true;
+	}
+
+	private void RebaseLine()
+	{
+		var rootPosition = arcGISMapComponent.GetComponent<HPRoot>().RootUniversePosition;
+		var delta = (lastRootPosition - rootPosition).ToVector3();
+		if (delta.magnitude > 1) // 1km
+		{
+			if (lineRenderer != null)
+			{
+				Vector3[] points = new Vector3[lineRenderer.positionCount];
+				lineRenderer.GetPositions(points);
+				for (int i = 0; i < points.Length; i++)
+				{
+					points[i] += delta;
+				}
+				lineRenderer.SetPositions(points);
+			}
+			lastRootPosition = rootPosition;
+		}
+	}
+
+	public void SetPolylineMode()
+	{
+		ResetMode();
+
+		isPolylineMode = true;
+		ModeButtons[0].interactable = false;
+
+		UnitButtons[0].GetComponentInChildren<TMP_Text>().text = "mi";
```

Comment:
Suggestion - Move these to reset. Then, when adding ^2 in the polygon and envelope methods, append using a loop.
```
foreach(var button in UnitButtons)
{
    UnitButtons[0].GetComponentInChildren<TMP_Text>().text += "<sup>2</sup>";
}
```

I think a mode enum and switch statements is probably the best way to write this, but not super important.

Reply:
Used switched and enums and I think the readability decreased a bit, I could switch to this if you think this is better. 

```
public enum DrawingMode
{
    Polyline,
    Polygon,
    Envelope
}

    switch (mode)
    {
        case DrawingMode.Polyline:
            isPolylineMode = true;
            SetUnitTexts(new[] { "mi", "ft", "m", "km" });
            ModeButtons[0].interactable = false;
            break;

        case DrawingMode.Polygon:
            isPolygonMode = true;
            SetUnitTextsWithSup2();
            ModeButtons[1].interactable = false;
            break;

        case DrawingMode.Envelope:
            isEnvelopeMode = true;
            SetUnitTextsWithSup2();
            ModeButtons[2].interactable = false;
            break;
    }
```

Reply:
The reason to hard code unit text is because if I grab the button text each time, say if I grab the "km^2", it would append another superscript which would have two superscripts and keep increasing. I could add a check but that just seem unnecessary. 

```
UnitButtons[0].GetComponentInChildren<TMP_Text>().text = "mi<sup>2</sup>";
UnitButtons[1].GetComponentInChildren<TMP_Text>().text = "ft<sup>2</sup>";
UnitButtons[2].GetComponentInChildren<TMP_Text>().text = "m<sup>2</sup>";
UnitButtons[3].GetComponentInChildren<TMP_Text>().text = "km<sup>2</sup>";
```

Reply:
Not very good suggestions I guess. Thanks for giving them a try.